### ISSUE TITLE
How to add JSON tags for parameters?

### ISSUE THREAD

**Issue by mayurpate** [go-swagger/go-swagger#1915]
## Problem statement
I am writing a swagger spec for designing API's. I am using this swagger spec to generate go client code by using "swagger generate client" command.  I have defined parameters for API's but generated code is not showing up the json tags. 
## Swagger specification
sample.yaml 
paths:
 /user/create:
   get:
       description: "some description"
       tags: "user"
       parameters:
           - x-go-name: "UserName"
             description: "some description"
             name: "username"
             in: "query"
             required: true
             type: "string"
     
## Steps to reproduce
1. swagger generate client -f sample.yaml -t <client_path>

This shows the following struct in *_parameters.go file:
type GetUserCreateParams struct {
// HTTP Request Object
HTTPRequest *http.Request `json:"-"`
/*
Some desc
Required: true
In: query
*/
Username string
}

In the above generated code, I am expecting the username parameter to be generated with json tag. Am I missing something in the swagger spec? 

## Environment
swagger version: 0.18.0
go version: go 1.12
OS:   Ubnutu 14.04

**Comment by fredbi** [comment]
No, query and path parameters do not lead to json tag generation.
Tags are generated with body params, responses and model definitions.

**Comment by mayurpate** [comment]
Thanks, @fredbi. Could you please clarify why json tags are not generated for a path and query params? 
Because if backend service is expecting a json tag then it would be really easy to map with actual field. Now in the current scenario, I have to manually map the json tags with actual field.

**Comment by fredbi** [comment]
These params are not passed in the HTTP request body, but in the URL. There is no json unmarshalling to be considered here, and there is no "backend expecting a json tag".
